File: feature/feature-template/src/main/java/org/gampiot/robok/feature/template/code/java/JavaClassTemplate.java
package org.gampiot.robok.feature.template.code.java;

import android.os.Parcel;

import org.gampiot.robok.feature.template.code.CodeTemplate;

public class JavaClassTemplate extends CodeTemplate {

     public JavaClassTemplate() {}

     public JavaClassTemplate(Parcel in) {
          super(in);
     }
     
     @Override
     public String getName() {
          return "Java class";
     }
     
     @Override
     public void configure() {
          setContent(
               "package "
               + PACKAGE_NAME
               + ";\n"
               + "\npublic class "
               + CLASS_NAME
               + " {\n\t\n}"
          );
     }
     
     @Override
     public String getExtension() {
          return ".java";
     }
}

File: feature/feature-template/src/main/kotlin/org/gampiot/robok/feature/template/code/java/JavaClassTemplate.kt
package org.gampiot.robok.feature.template.code.java

/*
 *  This file is part of Robok © 2024.
 *
 *  Robok is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Robok is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *   along with Robok.  If not, see <https://www.gnu.org/licenses/>.
 */ 

import android.os.Parcel

import org.gampiot.robok.feature.template.code.CodeTemplate

open class JavaClassTemplate : CodeTemplate {

    constructor() : super()

    constructor(parcel: Parcel) : super(parcel)

    override fun getName(): String {
        return "Kotlin class"
    }

    override fun configure() {
        setContent(
            """
            package ${getCodeClassPackageName()}

            class ${getCodeClassName()} {

            }
            """.trimIndent()
        )
    }

    override fun getExtension(): String {
        return ".kt"
    }
}

